### PR TITLE
Add docs for 24.10.0-preview1

### PR DESCRIPTION
**Pull request by hayley-jean**
- What's new
- AutoScavenge plugin
- Stream policy plugin
- Encryption-at-rest plugin
- License plugin
- Replace PackageCloud with Cloudsmith

### CODE REVIEW

**Review by coloradomorrissey — COMMENTED** [review]
Hi Haley, I have one comment and a question about our enterprise features. They are potentially inconsequential, so hopefully, this doesn't delay your push.  

The Custom Stream Policies section reads, "::: note If a policy update is invalid, it will not be applied and an error will be logged. ESDB will continue running with the previous valid policy in place. :::"
_“ESDB” should be updated to “EventStoreDB” for consistency - I don’t see “ESDB” outside of code in other areas of the updates._  

The AutoScavenge Plugin & Encryption at Rest Plugin sections read, "This plugin is bundled in EventStoreDB version 24.10 but is disabled by default."
_Aren't both AutoScavenge and Encryption at Rest enterprise features?  Perhaps I’m not familiar with the semantics we use with the transition in licensing and differentiating opensource/enterprise._

**Comment by alexeyzimarev** [comment]
Can we hold this for a bit until we can discuss my Hope-based PR?

File: docs/security.md [hunk 1]
```diff
@@ -1013,3 +1277,103 @@ Signature Hash: 6d922badaba2372070f13c69b620286262eab1d8d2d2156a271a1d73aaaf64e4
...
+			}
+		}
+	}
+}
+```
+
+The `Transform` configuration parameter must be specified in the server's configuration file:
+
+```
+Transform: AesGcm
```
Comment (shaan1337) [inline docs/security.md:1351]:
```suggestion
Transform: aes-gcm
```

File: docs/security.md [hunk 2]
```diff
@@ -1013,3 +1277,103 @@ Signature Hash: 6d922badaba2372070f13c69b620286262eab1d8d2d2156a271a1d73aaaf64e4
...
 | Plugin enabled and user not authenticated | If the plugin has been enabled but there are still access denied errors, check the following: <ul><li>The user exists and is enabled in the EventStoreDB database. Can you log in with the username and password?</li><li>The user certificate is valid, and has a valid chain up to a trusted root CA.</li><li>The user certificate and node certificate share a common root CA.</li><li>Use 'requires leader' (which is the default) in your client configuration to rule out issues with forwarding requests.</li></ul> |
+
+## Encryption-At-Rest <Badge type="warning" vertical="middle" text="Commercial"/>
+
+The Encryption-At-Rest plugin allows users to encrypt their EventStoreDB database. Currently, only chunk files are encrypted - the indexes are not. The primary objective is to protect against an attacker who obtains access to the physical disk. In contrast to volume level encryption, file level encryption provides some degree of protection for attacks against the live system or remote exploits as the plaintext data is not directly readable. Protecting against memory-dump based attacks is out of the scope of this plugin.
```
Comment (shaan1337) [inline docs/security.md:1283]:
```suggestion
The Encryption-At-Rest plugin allows users to encrypt their EventStoreDB database. Currently, only chunk files are encrypted - the indexes are not. The primary objective is to protect against an attacker who obtains access to the physical disk. In contrast to volume or filesystem encryption, file level encryption provides some degree of protection for attacks against the live system or remote exploits as the plaintext data is not directly readable. Protecting against memory-dump based attacks is out of the scope of this plugin.
```

File: docs/security.md [hunk 3]
```diff
@@ -1013,3 +1277,103 @@ Signature Hash: 6d922badaba2372070f13c69b620286262eab1d8d2d2156a271a1d73aaaf64e4
...
+
+### Encryption Algorithm
+
+Data is encrypted using a symmetric-key encryption algorithm.
+
+
+The plugin is designed to support different encryption algorithms. Currently, the following encryption algorithms are implemented:
+- `AesGcm` - Advanced Encryption Standard (AES) Galois Counter Mode (GCM)
+  - The default key size is 256 bits, but 128-bit and 192-bit keys are also supported.
+  - FIPS 140-2 compliant
```
Comment (shaan1337) [inline docs/security.md:1293]:
```suggestion
```

File: docs/security.md [hunk 4]
```diff
@@ -1013,3 +1277,103 @@ Signature Hash: 6d922badaba2372070f13c69b620286262eab1d8d2d2156a271a1d73aaaf64e4
...
+}
+```
+
+The `Transform` configuration parameter must be specified in the server's configuration file:
+
+```
+Transform: AesGcm
+```
+
+When using the `File` master key source, a master key must be generated using [es-genkey-cli](https://github.com/EventStore/es-genkey-cli) and placed in the configured `KeyPath` directory. The master key having the highest ID will be chosen as the active one.
```
Comment (shaan1337) [inline docs/security.md:1354]: the repository is currently private

File: docs/security.md [hunk 5]
```diff
@@ -1013,3 +1277,103 @@ Signature Hash: 6d922badaba2372070f13c69b620286262eab1d8d2d2156a271a1d73aaaf64e4
...
+				"Encryption": {
+					"AesGcm": {
+						"Enabled": true,
+						"KeySize": 256 # optional. supported key sizes: 128, 192, 256 (default)
+					}
+				}
+			}
+		}
+	}
+}
```
Comment (shaan1337) [inline docs/security.md:1345]:
```suggestion
{
  "EventStore": {
    "Plugins": {
      "EncryptionAtRest": {
        "Enabled": true,
        "MasterKey": {
          "File": {
            "KeyPath": "/path/to/keys/"
          }
        },
        "Encryption": {
          "AesGcm": {
            "Enabled": true,
            "KeySize": 256 # optional. supported key sizes: 128, 192, 256 (default)
          }
        }
      }
    }
  }
}
```

File: docs/security.md (hunk 4 above)
Comment (hayley-jean) [reply docs/security.md]: Thanks, I've removed the link for now

File: docs/server/configuration/license-keys.md [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+Environment variable:
+
+```
+EventStore__Plugins__Licensing__LicenseKey={Your key}
+```
+
+Configuration file:
+
+```
+{
```
Comment (alexeyzimarev) [inline docs/server/configuration/license-keys.md:21]: Can we please stick to two spaces indentation for JSON files and snippets?

File: docs/server/configuration/security.md [hunk 7]
```diff
@@ -716,6 +716,270 @@ access to `$settings` unless you specifically override it.
...
+
+```
+[31124,16,11:18:15.099,DBG] StreamBasedPolicySelector Successfully wrote default policy to stream $policies.
+```
+
+The default policy looks like this:
+
+```
+{
+	"streamPolicies": {
```
Comment (alexeyzimarev) [inline docs/server/configuration/security.md:782]: Same here, indentation.

File: docs/server/configuration/license-keys.md (hunk 6 above)
Comment (alexeyzimarev) [reply docs/server/configuration/license-keys.md:21]: I think we should move it to the installation instructions.

**Review by timothycoleman — APPROVED** [review]

**Comment by alexeyzimarev** [comment]
Can we include connectors HTTP API to the API reference docs? I assume those endpoints should be included to the swagger file?

**Comment by hayley-jean** [comment]
> Can we include connectors HTTP API to the API reference docs? I assume those endpoints should be included to the swagger file?

I think we can do that as part of moving the connectors docs from [EventStore/documentation](https://github.com/EventStore/documentation) to this repo

**Comment by alexeyzimarev** [comment]
> I think we can do that as part of moving the connectors docs from [EventStore/documentation](https://github.com/EventStore/documentation) to this repo

That's already done

**Review by timothycoleman — APPROVED** [review]

File: docs/server/configuration/plugins.md [hunk 8]
```diff
@@ -5,7 +5,9 @@ order: 6
 
 # Plugins configuration
 
-The commercial edition of EventStoreDB ships with several plugins that augment the behavior of the open source server. Each plugin is documented in relevant sections. For example, under Security, you will find the User Certificates plugin documentation.
+EventStoreDB ships with several plugins that augment the behavior of the open source server. Each plugin is documented in relevant sections. For example, under Security, you will find the User Certificates plugin documentation.
```
Comment (ylorph) [inline docs/server/configuration/plugins.md:8]: > open source server
We are not open source anymore: we are source available
Open Source can only be used when  using an OSI approved licence 
I'm unsure what the correct term is , @alexeyzimarev  any idea ?

Comment (alexeyzimarev) [reply docs/server/configuration/plugins.md:8]: We discussed it, will be fixed.

**Pull request merged**